addons/common/fnc_headDir.sqf

```diff
@@ -54,49 +54,37 @@ Author:
 #include "script_component.hpp"
 SCRIPT(headDir);
 
-private["_azimuth", "_angle", "_dif", "_infov", "_object", "_threed", "_do"];
-
-params ["_unit"];
-_object = param [1,_unit];
-_threed = false;
-
-_do = (typeName _object != typeName _unit);
-if (!_do) then {
-    _do = (_object != _unit);
-};
-
-if (_do) then {
-    _angle = [(_unit call CBA_fnc_getPos),(_object call CBA_fnc_getPos)] call BIS_fnc_dirTo;
-} else {
-    _angle = 0;
-};
-if (_unit != player) then
-{
-    _azimuth = getdir _unit;
+params [
+    ["_unit", objNull, [objNull]],
+    ["_offsetObject", objNull]
+];
+
+private _azimuth = 0;
+private _isExternalCam = false;
+if (_unit != call CBA_fnc_currentUnit) then {
+    _azimuth = getDir _unit;
 } else {
-    private ["_position","_viewPos","_vector","_magnitude"];
-    _position = positionCameraToWorld [0, 0, 0];
-    _threed = if ((_position distance _unit)>2) then {true} else {false};
-    _viewPos = positionCameraToWorld [0, 0, 99999999];
-    _vector = [
-        (_viewPos select 0) - (_position select 0),
-        (_viewPos select 1) - (_position select 1),
-        (_viewPos select 2) - (_position select 2)
-    ];
-    _magnitude = [0, 0, 0] distance _vector;
-    _vector = [
-        (_vector select 0) / _magnitude,
-        (_vector select 1) / _magnitude,
-        (_vector select 2) / _magnitude
-    ];
+    private _camPos = AGLToASL positionCameraToWorld [0, 0, 0];
+    _isExternalCam = (_camPos vectorDistance getPosASL _unit) > 2;
+    private _viewPos = AGLToASL positionCameraToWorld [0, 0, 99999999];
+    private _vector = _viewPos vectorDiff _camPos;
+    _vector = _vector vectorMultiply (1 / vectorMagnitude _vector);
     _azimuth = (_vector select 0) atan2 (_vector select 1);
     _azimuth = [_azimuth] call CBA_fnc_simplifyAngle;

```

Comment:
81-86 are for after the offset is applied. Could move the simplify after the offset and remove 81-86.